File: core/src/main/java/com/xmartlabs/bigbang/core/Injector.java
package com.xmartlabs.bigbang.core;

import android.support.annotation.NonNull;

import bullet.ObjectGraph;

public final class Injector {
  private static Injector instance = new Injector();

  private ObjectGraph bullet;

  private Injector() {
  }

  public static <T> void inject(@NonNull T t) {
    if (instance.bullet == null) {
      throw new IllegalStateException("No ObjectGraph is present. Did you forgot to call setObjectGraph?");
    }
    instance.bullet.inject(t);
  }

  public static Injector getInstance() {
    return instance;
  }

  public void setObjectGraph(@NonNull ObjectGraph bullet) {
    this.bullet = bullet;
  }
}

File: core/src/main/java/com/xmartlabs/bigbang/core/Injector.kt
package com.xmartlabs.bigbang.core

import bullet.ObjectGraph

class Injector private constructor() {
  var bullet: ObjectGraph? = null

  companion object {
    @JvmStatic
    val instance by lazy { Injector() }

    @JvmStatic
    fun <T> inject(t: T) {
      if (instance.bullet == null) {
        throw IllegalStateException("No ObjectGraph is present. Did you forgot to call setObjectGraph?")
      }
      instance.bullet?.inject(t)
    }
  }
}
